packages/core/src/storage/vectorStore/ChromaVectorStore.ts

```diff
@@ -0,0 +1,133 @@
+import { VectorStore, VectorStoreQuery, VectorStoreQueryMode, VectorStoreQueryResult } from "./types";
+import { BaseNode, Document, MetadataMode } from "../../Node";
+import { ChromaClient, Collection, IncludeEnum, Where, WhereDocument, Metadata, Embeddings } from 'chromadb'
+
+type ChromaDBParams = {
+  collectionName: string
+}
+
+type ChromaDeleteOptions = {
+  where?: Where;
+  whereDocument?: WhereDocument;
+}
+
+type ChromaQueryOptions = {
+  whereDocument?: WhereDocument
+}
+
+export class ChromaVectorStore implements VectorStore {
+  storesText: boolean = true;
+  private chromaClient: ChromaClient
+  private collection: Collection | null;
+
+  constructor(params: ChromaDBParams) {
+    this.chromaClient = new ChromaClient();
+    this.collection = null;
+    this.chromaClient.createCollection({
+      name: params.collectionName,
+    }).then((collection) => {
+      this.collection = collection;
+    }).catch((err) => {
+      console.error(err);
+    }
+    );
+  }
+
+  client(): ChromaClient {
+    return this.chromaClient;
+  }
+
+  async add(embeddingResults: BaseNode[]): Promise<string[]> {
+    if (!this.collection) {
+        throw new Error("Must initialize collection before adding.");
+    }
+
+    // Check if all nodes have an embedding and only if they do, add the embeddings to the collection
+    const allNodesHaveEmbeddings = embeddingResults.every((node) => node.embedding != undefined);
+    const embeddings = allNodesHaveEmbeddings ? embeddingResults.map((node) => node.getEmbedding()): undefined
+    try {
+      await this.collection.add({
+        ids: embeddingResults.map((node) => node.id_),
+        metadatas: embeddingResults.map((node) => node.metadata),
```

Comment:
@yisding @logan-markewich this is different than `chroma.py`: 

It calls `node_to_metadata_dict`, so it stores the node's content in the metadata AND also stores `node.getContent(MetadataMode.NONE)` in `documents`

Reply:
So for chroma in python, we store the node in the metadata, but without the text

```python
metadata_dict = node_to_metadata_dict(
    node, remove_text=True, flat_metadata=self.flat_metadata
)
```

This is basically so that we can easily reconstruct the complete node when retrieving

```python
node = metadata_dict_to_node(metadata)
node.set_content(text)
```

Then, there is some fallback method during retrieval, to support vector dbs that weren't created with llama-index